core/src/main/java9/com/linecorp/armeria/internal/Java9VersionSpecific.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.internal;
+
+import java.time.Clock;
+import java.time.Instant;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Implementation of {@link JavaVersionSpecific} using Java 9 APIs.
+ */
+class Java9VersionSpecific extends JavaVersionSpecific {
+
+    @Override
+    public final long currentTimeMicros() {
```

Comment:
remove `final` to give a change override by the next version of Java?

Reply:
While I think the inheritance will help in general, I figured having one step to opt-in to overriding a method (remove final) e.g., in Java 10 is a good way to avoid accidental overriding which I guess you were worried about too :)

Reply:
OK, That's a good idea! 👍